site/content/en/docs/Guides/development-server.md

```diff
@@ -0,0 +1,38 @@
+---
```

Comment:
Let's add a `publishDate: 2018-02-19` to the top content so that it doesn't show up before the stable release comes out.

@ReDucTor can you take a look, I'm not sure the phrasing here is right. Maybe it should be something more like:
"Coordinate Agones with a local development server" or something? I'm not quite sure what the documentation should say -- I figured you might have some good words.

We aren't actually hosting it in Agones - we're more coordinating it with a local build?

Reply:
It also turns out that we need a bunch of the pod spec template stuff to make this all work I'm adding comments for now to say it's ignored. I've been trying to experiment with not requiring it but I have to rip out a lot of validation that's in the yaml files to inch towards that. I'm now experimenting with just putting placeholder values for now.

Reply:
Saying it's ignored seems reasonable :+1:

Reply:
The only thing that causes a bit of confusion I believe is the phrase "Development mode" which makes it seem like its a feature toggle within all of Agones (even though it is in some ways). An alternative might be "Development game server"

Reply:
What should this feature be called in the docs?

* Development Mode
* Unmanaged Mode
* Local Mode

etc?

Reply:
Good news: I got the PodSpecTemplate to be optional in my experimental branch.
Bad news: It requires a bug fix in jsonpatch to get it working. I'll see if I put up a PR for that project.

Reply:
This is because we define it as required in the OpenAPI spec, yeah?

Reply:
Could we call the document "Overwrite GameServer IP and Port" ? - which is literally what it is?

And then in the document, outline some use case(s) - such as running a development game server on your local machine, but exposing it to you QA Agones cluster by registering it in this special way - ? WDYT?

Reply:
I'd put a vote for "unmanaged mode".
I think it's clearer to align with a use case (intent) rather than explain what the logic is doing.

Unmanaged is generic enough to cover these cases like the QA workflow.

Reply:
On the topic of jsonpatching and clearing out all the extra fields.

Also I looked into the json patch stuff. It's actually a bit more complicated than originally thought and probably past being worth the time now. It looks like a subtle bug between the 2 different json libraries that we use. I think it's just better to move away from the mattbierd library and focus on using the evanphx one that Kubernetes already depends on.

Reply:
AFAIK evanphx will not generate patches. Unless I missed something?

Reply:
https://github.com/evanphx/json-patch#create-and-apply-a-json-patch

Reply:
This is a "Guide" - which is a howto, essentially, so thinking of it as an action - i.e. what is the user trying to do here?

It doesn't need a "mode" per se - it just needs to describe what we are doing.

- "Inject local Game Server into Agones"
- "Point Agones at local game server"
- "Manage local game server with Agones"

Something like that?

Reply:
Register Local Game Server with Agones?

Reply:
I like it! :+1:

Reply:
Updated